pkg/controller/controller_instance.go

```diff
@@ -303,15 +337,43 @@ func (c *controller) reconcileInstance(instance *v1alpha1.Instance) error {
 	glog.V(4).Infof("Provisioning a new Instance %v/%v of ServiceClass %v at Broker %v", instance.Namespace, instance.Name, serviceClass.Name, brokerName)
 	response, err := brokerClient.ProvisionInstance(request)
 	if err != nil {
+		// There are two buckets of errors to handle:
+		// 1.  Errors that represent a failure response from the broker
+		// 2.  All other errors
+		if osb.IsHTTPError(err) {
+			// An error from the broker represents a permanent failure and
+			// should not be retried; set the Failed condition.
+
+			httpErr := err.(osb.HTTPStatusCodeError)
+			s := fmt.Sprintf("Error provisioning Instance \"%s/%s\" of ServiceClass %q at Broker %q: %s", instance.Namespace, instance.Name, serviceClass.Name, brokerName, httpErr)
+			glog.Warning(s)
+			clone, err := api.Scheme.DeepCopy(instance)
+			if err != nil {
+				return err
+			}
+			toUpdate := clone.(*v1alpha1.Instance)
+
+			setInstanceCondition(toUpdate, v1alpha1.InstanceConditionFailed, v1alpha1.ConditionTrue, "BrokerReturnedFailure", s)
+			c.updateInstanceCondition(
+				instance,
```

Comment:
Is this persisting the `InstanceConditionFailed` condition? You cloned `instance` into `toUpdate`, set the `InstanceConditionFailed` on `toUpdate`, but then passed the original `instance` in here. I would think this would need to be `toUpdate` to work.